modules/apigee_edge_teams/src/Plugin/Block/TeamContextSwitcherBlock.php

```diff
@@ -0,0 +1,145 @@
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The team membership manager.
+   *
+   * @var \Drupal\apigee_edge_teams\TeamMembershipManagerInterface
+   */
+  protected $teamMembershipManager;
+
+  /**
+   * The team entity storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $storage;
+
+  /**
+   * The form builder.
+   *
+   * @var \Drupal\Core\Form\FormBuilderInterface
+   */
+  protected $formBuilder;
+
+  /**
+   * TeamContextSwitcher constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\apigee_edge_teams\TeamMembershipManagerInterface $team_membership_manager
+   *   The Apigee team membership manager.
+   * @param \Drupal\Core\Entity\EntityStorageInterface $storage
+   *   The team entity storage.
+   * @param \Drupal\Core\Form\FormBuilderInterface $form_builder
+   *   The form builder.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, AccountInterface $current_user, TeamMembershipManagerInterface $team_membership_manager, EntityStorageInterface $storage, FormBuilderInterface $form_builder) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->currentUser = $current_user;
+    $this->teamMembershipManager = $team_membership_manager;
+    $this->storage = $storage;
+    $this->formBuilder = $form_builder;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_user'),
+      $container->get('apigee_edge_teams.team_membership_manager'),
+      $container->get('entity_type.manager')->getStorage('team'),
+      $container->get('form_builder')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function blockAccess(AccountInterface $account) {
+    return AccessResult::allowedIf($account->isAuthenticated());
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // If the user can administer team, show all teams.
+    if ($this->currentUser->hasPermission('administer team')) {
+      $teams = NULL;
+    }
+    // Otherwise show teams the user belongs to.
+    elseif (!($teams = $this->teamMembershipManager->getTeams($this->currentUser->getEmail()))) {
+      return NULL;
+    }
+
+    return $this->formBuilder->getForm(TeamContextSwitcherForm::class, $this->storage->loadMultiple($teams));
```

Comment:
What should be displayed to a user if it does not have access to any team? The form with the empty "Select a team" select list?

Reply:
No block is displayed if the user does not have access to any team.